Lib9c/Action/Coupons/TransferCoupons.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+
+namespace Nekoyume.Action.Coupons
+{
+    [Serializable]
+    [ActionType("transfer_coupons")]
+    public sealed class TransferCoupons : GameAction
+    {
+        public TransferCoupons(
+            IImmutableDictionary<Address, IImmutableSet<Guid>> couponsPerRecipient)
+        {
+            CouponsPerRecipient = couponsPerRecipient;
+        }
+
+        public IImmutableDictionary<Address, IImmutableSet<Guid>> CouponsPerRecipient
+        {
+            get;
+            private set;
+        }
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            var states = context.PreviousStates;
```

Comment:
Could you handle the case when `context.Rehearsal` is `true`?
And I usually use the pattern below.
```suggestion
        public override IAccountStateDelta Execute(IActionContext context)
        {
            var states = context.PreviousStates;
            if (context.Rehearsal)
            {
                return states;
            }
```

Reply:
We've ensured the current action code to work well with rehearsal mode, and to calculate more accurate `UpdatedAddresses`.

Reply:
All right, Thanks for the explanation.